### PR TITLE
Handle Samsung X_SetBookmark action

### PR DESCRIPTION
**Pull request by dyseg**
I originally created these modifications only for myself (with my limited C++ skills). But since it is working, why not share it?
I stored the "bookmark position" in the existing mt_cds_object table in a new column. Maybe it is not ideal, considering not all TVs will support this, but the column will be there anyway. Also, I was only able to test with sqlite.
The implementation details are coming from the [Serviio blog post from 2011](https://github.com/gerbera/gerbera/issues/848#issuecomment-624485095).
All tests were done on UE50TU7102, which sends back the bookmark position as msec, although in he Serviio blog post the example shows that it is sent back as sec. Also it would be better to store the bookmark position on per client basis, maybe in a new SQL table.

### CODE REVIEW

**Comment by whyman** [comment]
This is great, thanks @dyseg!

I think it would be ideal to wire this into our client detection "tweaks" mechanism (at least for sending the saved position), but otherwise it looks nice.

**Comment by dyseg** [comment]
I've moved the X_SetBookMark action handling to the Quirks class. Please share your thoughts @whyman. Also the global config option to switch on/off the bookmark handing was removed, I think it's is not necessary any more, since we have to switch it on for every client explicitly anyway.

I would still like to store the bookmarks corresponding to single clients, I'll try to implement it as a next step.

**Comment by dyseg** [comment]
I've added a new quirk option, since older Samsung TVs send back the bookmark position as second(like in the original Serviio forum post). This was verified with UE43J5500. Also from now on, in the database the bookmark value is always stored as 'second', and multiplied if necessary.

I think at this point You should merge this PR, and later on if I have time I'll finish the per client based bookmark storing. By the way, maybe it's not even necessary, I found it useful to stop on one TV and continue on the other :).

Also, please mark that this PR will resolve #848

**Comment by KarlStraussberger** [comment]
There is a conflict in `src/upnp_xml.cc` which needs to be resolved before the PR can be merged

File: src/util/upnp_quirks.h
Comment (whyman) [inline src/util/upnp_quirks.h:34]: These should go with the other forward declarations below

File: src/upnp_cds.h
Comment (whyman) [inline src/upnp_cds.h:104]: I guess this is from the first version, looks like its not actually needed now?

File: src/upnp_xml.cc
Comment (whyman) [inline src/upnp_xml.cc:108]: Probably should be `nullptr`?

**Review by whyman — CHANGES_REQUESTED** [review]
Looks nice!

File: src/util/upnp_quirks.h
Comment (dyseg) [reply src/util/upnp_quirks.h]: Resolved in https://github.com/gerbera/gerbera/pull/1288/commits/cb16aa7074d22b899544f986df6b1d0821b2e66f

File: src/upnp_cds.h
Comment (dyseg) [reply src/upnp_cds.h]: Yes, you are right, this is not necessary any more. Reverted in: https://github.com/gerbera/gerbera/pull/1288/commits/fc6dc8807a616bbf64c663fc8a625cc62c0760d4

File: src/upnp_xml.cc
Comment (dyseg) [reply src/upnp_xml.cc]: Resolved in https://github.com/gerbera/gerbera/pull/1288/commits/e23aeff8f0cdb93e062b27881e206442902dbba2

**Comment by whyman** [comment]
LGTM, happy to merge, but there are conflicts

**Comment by dyseg** [comment]
> LGTM, happy to merge, but there are conflicts

Sorry but I don't see where. Locally auto merge is possible into gerbera:master, and Github also states the same. Am I missing something?

**Comment by whyman** [comment]
We generally do rebase merges.

We can squash merge without conflicts if you prefer?

**Comment by dyseg** [comment]
Squash merge is fine for me. Maybe it's even better to have all the changes in a single commit.

**Pull request merged**
